manopt/tools/checkdiff.m

```diff
@@ -114,5 +114,7 @@ function checkdiff(problem, x, d, force_gradient)
     fprintf('The slope should be 2. It appears to be: %g.\n', poly(1));
     fprintf(['If it is far from 2, then directional derivatives ' ...
              'might be erroneous.\n']);
+    fprintf(['If it is very close to 0, then the modeled funtion is '...
```

Comment:
Right!
Actually a linear model should be quite unusual (probably even more unusual than the quadratic one), and would be unbounded anyway, so the practical interest is quite low.
However, just for completeness, I think it's fine to have this special case considered as well (it also provides a more correct analysis considering the numerical approximation framework deployed for the test). This complements the theoretical exposition in the Manopt documentation and might came in handy for those just learning the mechanics.

Then, regarding the NaN you're right again. I've added an if-else structure, as in the case of the checkhessian, which simply checks if the numerical errors are low enough to point towards a potential exact model. I think 1e-12 should be low enough to not to produce false positives and still having a good margin for detection of exact models (where actual errors are close to eps, 1e-15 to 1e-16).